### PR TITLE
vector-based wind and emission entities

### PR DESCRIPTION
**Pull request by q009**
This change introduces vector-based wind entities and emitters.
The shader will now animate meshes in the direction of the wind, which is either provided by temporary emitters or map entities.

Introduced map variables:
windspeed(alt)
windyaw(alt)
windinterval(alt)

New map entity "wind"

Please consult the in-game info/docs for usage.

### CODE REVIEW

File: src/engine/wind.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    // apply the direction
+    vecfromyaw(checkmapvariant(MPV_ALT) ? windyawalt : windyaw, 1, 0, globalwind);
+    globalwind.mul(speed);
+}
+
+// gets the wind for dynamic (moving) entities
+static vec getentwindvec(const dynent *d)
+{
+    // add dynent velocity to the wind vector
+    return d ? vec(d->vel).add(d->falling).mul(-WIND_ENT_FALL_SCALE) : vec(0);
```
Comment (q009) [inline src/engine/wind.cpp]: Noticed this oopsie here, this is wrong

Comment (qreeves) [reply src/engine/wind.cpp]: That moment when people review their own pull requests 🤣

**Comment by qreeves** [comment]
You think you're able to add `wind.[cpp|o]` to the other project files (CBP, XCODEPROJ), or do you need me to do it?

File: src/engine/wind.cpp [hunk 2]
```diff
@@ -0,0 +1,255 @@
+#include "engine.h"
+#include "game.h"
+
+#define WIND_IMPULSE_DECAY_MUL 4.0f
+#define WIND_ENT_FALL_SCALE 0.01f
+#define WIND_ATTEN_SCALE 0.01f
+#define WIND_MAX_SPEED 2.0f
+#define WIND_MAX_EMITTERS 100
```
Comment (qreeves) [inline src/engine/wind.cpp]: Is this stuff that could be a variable?

Comment (q009) [reply src/engine/wind.cpp]: I don't think exposing those to the user would provide any value.

**Comment by q009** [comment]
> You think you're able to add `wind.[cpp|h|o]` to the other project files (CBP, XCODEPROJ), or do you need me to do it?

Projects updated

**Pull request merged**